cmd/kubeadm/app/phases/copycerts/copycerts.go

```diff
@@ -261,14 +266,21 @@ func getSecret(client clientset.Interface) (*v1.Secret, error) {
 
 func getDataFromSecret(secret *v1.Secret, key []byte) (map[string][]byte, error) {
 	secretData := map[string][]byte{}
-	for certName, encryptedCert := range secret.Data {
-		cert, err := cryptoutil.DecryptBytes(encryptedCert, key)
-		if err != nil {
-			// If any of the decrypt operations fail do not return a partial result,
-			// return an empty result immediately
-			return map[string][]byte{}, err
+	for secretName, encryptedSecret := range secret.Data {
+		// In some cases the secret might have empty data if the secrets were not present on disk
+		// when uploading. This can specially happen with external insecure etcd (no certs), or
```

Comment:
please remove `, or with external certificates (e.g. missing ca.key when uploading the certs)`

In external CA mode, kubeadm cannot create the missing certificate. in fact, as discussed [here](https://github.com/kubernetes/kubernetes/pull/73907#discussion_r256426875) in the original PR in this case the user is always required to provide the certificates on the joining node manually -> the automatic copy certs doesn't make sense

This is also enforced in the code [here](https://github.com/kubernetes/kubernetes/blob/df2094b3d728bd58c5f23b41add109e32fc7c301/cmd/kubeadm/app/cmd/init.go#L346)

Reply:
Will remove that part of the comment but I don't understand why with external certs we cannot upload everything present to the secret and download on machines that join. If the ca.key is missing that's fine, but we can still transfer all secrets from one machine to another.

Of course not to be done now, but I wonder if I'm missing something here.